frontend/src/features/research/add/AddResearchForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { mdiFolderText } from '@mdi/js';
+import Icon from '@mdi/react';
+import { InlineInput } from 'components/common/form/styles';
+import * as Styled from 'components/common/styles';
+import { Formik } from 'formik';
+import { Api_ResearchFile } from 'models/api/ResearchFile';
+import * as React from 'react';
+import { Button, Col, Row } from 'react-bootstrap';
+import { Prompt, useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { useAddResearch } from '../hooks/useAddResearch';
+import { AddResearchFileYupSchema } from './AddResearchFileYupSchema';
+import { ResearchForm } from './models';
+import ResearchProperties from './ResearchProperties';
+
+const AddResearchForm: React.FunctionComponent = () => {
+  const initialForm = new ResearchForm();
+  const history = useHistory();
+
+  const { addResearchFile } = useAddResearch();
+
+  const submitForm = async (researchFile: Api_ResearchFile) => {
+    const response = await addResearchFile(researchFile);
+    if (!!response?.name) {
+      history.push(`/research/${response?.name}`);
+    }
+  };
+
+  const onCancel = () => {
+    history.push(`/`);
+  };
+  return (
+    <Formik<ResearchForm>
+      initialValues={initialForm}
+      onSubmit={async (values: ResearchForm, formikHelpers) => {
+        const researchFile: Api_ResearchFile = values.toApi();
+        formikHelpers.setSubmitting(false);
```

Comment:
This should possibly be after the submitForm call, and you may want to await the result of that call.

Reply:
Good to know. I think I copied this from another onSubmit